dist/performance.js

```diff
@@ -11,34 +23,106 @@ function getLcpElement() {
             resolve(naiveLcpEntry);
         }).observe({ type: "largest-contentful-paint", buffered: true });
     }).then(({ startTime, element, url, size, loadTime, renderTime }) => {
-        let attributes = [];
-        for (let index = 0; index < element.attributes.length; index++) {
-            const ele = element.attributes.item(index);
-            attributes[index] = { name: ele.name, value: ele.value };
-        }
-
+        const attributes = getAttributes(element);
+        const styles = getComputedStyles(element, ['background-image']);
         return {
             startTime,
-            nodeName: element.nodeName,
+            nodeName: element?.nodeName,
             url,
             size,
             loadTime,
             renderTime,
             attributes,
+            boundingClientRect: element?.getBoundingClientRect().toJSON(),
+            naturalWidth: element?.naturalWidth,
+            naturalHeight: element?.naturalHeight,
+            styles
```

Comment:
To check my understanding - here we are creating a way to get the URL when an LCP image is added via the `background-image` CSS rule?

Reply:
Yes! The `url` field above will be the LCP image even for DIVs with background image styles, but in those cases we don't necessarily know that it was that background image that was the LCP. This helps to clarify that.